frontend/src/components/elements/PlotlyChart/CustomTheme.tsx

```diff
@@ -0,0 +1,613 @@
+/**
+ * Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { merge, assign } from "lodash"
+
+import { useTheme } from "@emotion/react"
+
+import { hasLightBackgroundColor, Theme } from "src/theme"
+
+// TODO: for these colors below, these likely need to move to our theme!
```

Comment:
We probably can start thinking about putting the colors and some of the functions into shared utility. Especially if you will also pick up some theme fixes for Altair.

Reply:
Yep! I was thinking of doing that in a separate PR since I don't want to make the changes on altair in this PR. Is that ok to do in a separate PR?

Reply:
Sounds good 👍